pkgs/servers/nextcloud/packages/apps/recognize.nix

```
@@ -0,0 +1,136 @@
+        hash = currentVersionInfo.appHash;
+      })
+
+      (fetchurl {
+        inherit version;
+        url = "https://github.com/nextcloud/recognize/archive/refs/tags/v${version}.tar.gz";
+        hash = currentVersionInfo.modelHash;
+      })
+    ]
+    ++ lib.optionals useLibTensorflow [
+      (fetchurl rec {
+        # For version see LIBTENSORFLOW_VERSION in https://github.com/tensorflow/tfjs/blob/master/tfjs-node/scripts/deps-constants.js
+        version = "2.9.1";
+        url = "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-${version}.tar.gz";
+        hash = "sha256-f1ENJUbj214QsdEZRjaJAD1YeEKJKtPJW8pRz4KCAXM=";
+      })
+    ];
+
+  unpackPhase =
+    ''
+      # Merge the app and the models from github
+      tar -xzpf "${builtins.elemAt srcs 0}" recognize;
+      tar -xzpf "${builtins.elemAt srcs 1}" -C recognize --strip-components=1 recognize-${version}/models
+    ''
+    + lib.optionalString useLibTensorflow ''
+      # Place the tensorflow lib at the right place for building
+      tar -xzpf "${builtins.elemAt srcs 2}" -C recognize/node_modules/@tensorflow/tfjs-node/deps
+    '';
+
+  patchPhase = ''
+    # Make it clear we are not reading the node in settings
+    sed -i "/'node_binary'/s:'""':'Nix Controled':" recognize/lib/Service/SettingsService.php
+
+    # Replace all occurences of node (and check that we actually remved them all)
+    test "$(grep "get[a-zA-Z]*('node_binary'" recognize/lib/**/*.php | wc -l)" -gt 0
+    substituteInPlace recognize/lib/**/*.php \
+      --replace-quiet "\$this->settingsService->getSetting('node_binary')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary', '""')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary', '""')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary')" "'${nodejs}/bin/node'"
+    test "$(grep "get[a-zA-Z]*('node_binary'" recognize/lib/**/*.php | wc -l)" -eq 0
+
+
+
+    # Skip trying to install it... (less warnings in the log)
+    sed  -i '/public function run/areturn ; //skip' recognize/lib/Migration/InstallDeps.php
+
+    ln -s ${ffmpeg}/bin/ffmpeg recognize/node_modules/ffmpeg-static/ffmpeg
+  '';
+
+  buildInputs = [
+    nodejs
+    nodejs.pkgs.node-pre-gyp
+    nodejs.pkgs.node-gyp
+    python311
+    util-linux
+  ];
+  buildPhase = lib.optionalString useLibTensorflow ''
+    cd recognize
+
+    # Install tfjs dependency
+    export CPPFLAGS="-I${nodejs}/include/node -Ideps/include"
+    cd node_modules/@tensorflow/tfjs-node
+    node-pre-gyp install --prefer-offline --build-from-source --nodedir=${nodejs}/include/node
+    cd -
+
+    # Test tfjs returns exit code 0
+    ${nodejs}/bin/node src/test_libtensorflow.js
```

```suggestion
    node src/test_libtensorflow.js
```
Please move node to nativeBuildInputs

---

`nodejs` still needs to be moved to `nativeBuildInputs`.

---

Done :slightly_smiling_face: